sherlockml/clients/report.py

```diff
@@ -0,0 +1,141 @@
+    ],
+)
+
+
+class ReportVersionSchema(Schema):
+    id = fields.UUID(data_key="version_id", required=True)
+    created_at = fields.DateTime(required=True)
+    author_id = fields.UUID(required=True)
+    report_path = fields.String(required=True)
+    report_key = fields.String(required=True)
+    report_bucket = fields.String(required=True)
+    notebook_path = fields.String(required=True)
+
+    @post_load
+    def make_report_version(self, data):
+        return ReportVersion(**data)
+
+
+class ReportSchema(Schema):
+    created_at = fields.DateTime(required=True)
+    name = fields.String(required=True, data_key="report_name")
+    id = fields.UUID(required=True, data_key="report_id")
+    description = fields.String(required=True)
+    active_version = fields.Nested(ReportVersionSchema, required=True)
+
+    @post_load
+    def make_active_report(self, data):
+        return Report(**data)
+
+
+class ReportWithVersionsSchema(Schema):
+    created_at = fields.DateTime(required=True)
+    name = fields.String(required=True, data_key="report_name")
+    id = fields.UUID(required=True, data_key="report_id")
+    description = fields.String(required=True)
+    active_version_id = fields.UUID(required=True)
+    versions = fields.Nested(ReportVersionSchema, required=True, many=True)
+
+    @post_load
+    def make_versioned_report(self, data):
+        return ReportWithVersions(**data)
+
+
+class ReportClient(BaseClient):
+
+    SERVICE_NAME = "tavern"
+
+    def list(self, project_id):
+        endpoint = "/project/{}".format(project_id)
+        return self._get(endpoint, ReportSchema(many=True))
+
+    def get_with_versions(self, report_id):
+        endpoint = "/report/{}/versions".format(report_id)
+        return self._get(endpoint, ReportWithVersionsSchema())
+
+    def get(self, report_id):
+        endpoint = "/report/{}/active".format(report_id)
+        return self._get(endpoint, ReportSchema())
+
+    def create(
+        self,
+        project_id,
+        name,
+        notebook_path,
+        author_id,
+        description=None,
+        show_code=False,
+    ):
+
+        payload = {
+            "report_name": name,
+            "author_id": str(author_id),
+            "notebook_path": str(notebook_path),
+            "description": description,
+            "show_input_cells": show_code,
+        }
+
+        endpoint = "/project/{project_id}".format(project_id=project_id)
+        return self._post(endpoint, ReportSchema(), json=payload)
+
+    def create_version(
+        self, report_id, notebook_path, author_id, show_code=False
+    ):
+
+        payload = {
+            "notebook_path": str(notebook_path),
+            "author_id": str(author_id),
+            "show_input_cells": show_code,
+            "draft": False,
+        }
+
+        endpoint = "/report/{report_id}/version".format(report_id=report_id)
+        return self._post(
+            endpoint, ReportVersionSchema(unknown=EXCLUDE), json=payload
```

Comment:
I think `unknown=EXCLUDE` is the default behaviour - can we exclude this option here?